src/main/java/UI.java

```diff
@@ -0,0 +1,174 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UI {
+
+    private Scanner sc;
+    private List<Task> taskList;
+
+    private static String initGreeting = "Hello! I'm Duke \nWhat can I do for you?";
+
+    public UI() {
+        this.sc = new Scanner(System.in);
+        this.taskList = new ArrayList<Task>();
+    }
+
+    public String greet() {
+        return initGreeting;
+    }
+
+    /**
+     * Returns the current task list if it is not empty, prints an alternative message if otherwise.
+     *
+     * @return String which content depends on whether the current task list is empty
+     */
+    public String list() {
+
+        if (this.taskList.isEmpty()) {
+            return "No tasks as of now :)";
+        } else {
+            String temp = "";
+            for (int i = 0; i < taskList.size(); i++) {
+                int curr = i + 1;
+                temp = temp + curr + ": " + taskList.get(i).toString() + "\n";
+            }
+            return "Here are the tasks in your list:\n"
+                   + temp;
+        }
+    }
+
+    /**
+     * Returns a farewell message before Duke closes.
+     *
+     * @return String containing the farewell message
+     */
+    public String bye() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Adds the given task to the current task list.
+     *
+     * @param str String input specifying the type of task and its description.
+     * @return String telling the user what task was added and the number of current tasks.
+     * @throws DukeInvalidDescriptionException if the description of the given is empty.
+     */
+    public String addTask(String str) throws DukeInvalidDescriptionException {
```

Comment:
I like how you abstracted the commands into different methods, which will definitely help you in the subsequent methods. 👍